packages/core/src/browser/consoleObservable.ts

```diff
@@ -0,0 +1,84 @@
+import { callMonitored } from '../domain/internalMonitoring'
+import { computeStackTrace } from '../domain/tracekit'
+import { createHandlingStack, formatErrorMessage, toStackTraceString } from '../tools/error'
+import { mergeObservables, Observable } from '../tools/observable'
+import { find, jsonStringify } from '../tools/utils'
+
+export const ConsoleApiName = {
+  log: 'log',
+  debug: 'debug',
+  info: 'info',
+  warn: 'warn',
+  error: 'error',
+} as const
+
+export type ConsoleApiName = typeof ConsoleApiName[keyof typeof ConsoleApiName]
+
+export const CONSOLE_APIS = Object.keys(ConsoleApiName) as ConsoleApiName[]
+
+export interface ConsoleLog {
+  message: string
+  apiName: ConsoleApiName
+  stack?: string
+  handlingStack?: string
+}
+
+const consoleObservables: { [k in ConsoleApiName]?: Observable<ConsoleLog> } = {}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Can you add a test for the need to have a singleton?

Reply:
Isn't the "should allow multiple caller" test enough?

Reply:
🤔 removing the singleton does not fail the test, so something is not right

Reply:
Its because the observable references of `consoleObservablesByApi` live inside the merged observable.
I've updated the test to ensure the console api as been instrumented only once.
https://github.com/DataDog/browser-sdk/blob/18b2dbe7a138f0dbf50abc4ee3115801c6ac514e/packages/core/src/domain/console/consoleObservable.spec.ts#L65